mycore-base/src/main/java/org/mycore/common/xml/MCRXMLFunctions.java

```diff
@@ -1031,4 +1035,48 @@ public static String toNCNameSecondPart(String name) {
         return name.replaceAll("[^\\w\\-\\.]*", "");
     }
 
+    /**
+     * This only works with text nodes
```

Comment:
javadoc is no help. It should clearly state what is done with which input. If it's only working with text nodes, throw an Exception if a node is not of that type.

Reply:
I don't want to throw a exception, because you can not handle them in XSL.